OpenRA.Mods.Common/Traits/Buildings/Building.cs

```diff
@@ -164,7 +164,7 @@ public Actor FindBaseProvider(World world, Player p, CPos topLeft)
 
 				// Range is counted from the center of the actor, not from each cell.
 				var target = Target.FromPos(bp.Actor.CenterPosition);
-				if (target.IsInRange(center, bp.Trait.Info.Range))
+				if (target.IsInRange(center, bp.Trait.Info.Range) || !world.WorldActor.Trait<MapBuildRadius>().BuildRadiusEnabled)
```

Comment:
We already query the `MapBuildRadius` trait above and should 'cache' it therefore, imho.
I.e. something like:
```
public Actor FindBaseProvider(World world, Player p, CPos topLeft)
{
	var center = world.Map.CenterOfCell(topLeft) + CenterOffset(world);
	var mapBuildRadius = world.WorldActor.Trait<MapBuildRadius>();
	var allyBuildEnabled = mapBuildRadius.AllyBuildRadiusEnabled;

	foreach (var bp in world.ActorsWithTrait<BaseProvider>())
	{
		[...]

		if (mapBuildRadius.BuildRadiusEnabled || target.IsInRange(center, bp.Trait.Info.Range))
			return bp.Actor;
	}

	return null;
}
```